pwiz_tools/Skyline/TestUtil/TestFunctional.cs

```diff
@@ -689,7 +689,15 @@ public static void WaitForClosedForm(Form formClose)
                     {
                         // Grab some details in case of eventual failure
                         var formCloseClassName = System.ComponentModel.TypeDescriptor.GetClassName(formClose);
-                        var formCloseText = formClose.Text;
+                        string formCloseText;
+                        try
+                        {
+                            formCloseText = formClose.Text;
+                        }
+                        catch
+                        {
+                            formCloseText = "@@(could not retrieve form text)@@";
```

Comment:
Probably not worth the extra effort, but you could also set a flag that would cause you to try to formClose.Invoke() to get its text while on its own thread, rather than while on the main window thread.